nncf/torch/graph/patterns.py

```diff
@@ -11,28 +11,105 @@
  limitations under the License.
 """
 
-from nncf.torch.graph.graph import NNCFNodeExpression as N
+import copy
 from nncf.torch.graph.version_agnostic_op_names import VersionAgnosticNames
 
-LINEAR_OPS = N('linear') | N('conv2d') | N('conv_transpose2d') | N('conv3d') | \
-             N('conv_transpose3d') | N('conv1d') | N('addmm')
+import networkx as nx
+import networkx.algorithms.isomorphism as ism
 
-RELU = N(VersionAgnosticNames.RELU) | N('hardtanh')
+NODE_COUNTER = 0
 
-BN = N('batch_norm') | N('batch_norm3d')
 
-POOLING = N('adaptive_avg_pool2d') | N('adaptive_avg_pool3d') | N('avg_pool2d') | N('avg_pool3d')
+class GraphPattern:
+    def __init__(self, pattern_name, types):
```

Comment:
Unnecessary to store `pattern_name` in the pattern object.

Reply:
Done